### PR TITLE
Add confirm box to replace or add data.

### LINKED ISSUE #75: "Untitled" column
Probably some off-by-one error.

### LINKED ISSUE #78: Need to be able to handle replayed matches
They replayed match 30 today because of a field fault. There was no way for them to resubmit without causing conflicts when we compile the data.

### PR DESCRIPTION
**Pull request by AndrewLester**
Closes #78 
Closes #75

### CODE REVIEW

**Comment by AndrewLester** [comment]
2 Questions:

Is there a way to customize dialogue box text? (So it could be "no" instead of cancel)
Do you want the team number to also be the same when it checks if you are submitting the same match? (Current way)

**Comment by ErikBoesen** [comment]
Not without jQuery, and I don't want to use jQuery for only one purpose.

File: script.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ var tags = document.querySelectorAll('input:not(.special), select:not(.special),
 // Create empty object.
 var inputs = {};
 // Make each element be the value to a key named after its ID.
-for (t in tags) inputs[tags[t].id] = tags[t];
+for (t of tags.keys()) inputs[tags[t].id] = tags[t];
```
Comment (ErikBoesen) [inline script.js]: If you're using for/of, you wouldn't want to use `tags[t]`, would you?

File: script.js [hunk 2]
```diff
@@ -80,6 +80,10 @@ pg.submit.onclick = function() {
 
 function write(match) {
     var data = (fs.existsSync(path) && fs.statSync(path).size > 0) ? JSON.parse(fs.readFileSync(path)) : [];
+    for (var i = 0; i < data.length; i++) {
+        var replace = (data[i].match === match.match && data[i].team === match.team) ? confirm('Do you want to replace the previous data for this match?') : '';
```
Comment (ErikBoesen) [inline script.js]: Better to use an if here, no?

File: script.js (hunk 1 above)
Comment (AndrewLester) [reply script.js]: I'm looping through the keys though, so I need to plug the key back in right? (Now that I think about it I could loop through the values)

File: script.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ var tags = document.querySelectorAll('input:not(.special), select:not(.special),
 // Create empty object.
 var inputs = {};
 // Make each element be the value to a key named after its ID.
-for (t in tags) inputs[tags[t].id] = tags[t];
+for (t of tags.values()) inputs[t.id] = t;
```
Comment (ErikBoesen) [inline script.js]: I think using `tag` rather than `t` would be more consistent with the next few lines.

Comment (ErikBoesen) [reply script.js]: Also use `of` for the next one if you want.

File: script.js [hunk 4]
```diff
@@ -80,6 +80,13 @@ pg.submit.onclick = function() {
 
 function write(match) {
     var data = (fs.existsSync(path) && fs.statSync(path).size > 0) ? JSON.parse(fs.readFileSync(path)) : [];
+    for (var i = 0; i < data.length; i++) {
+        var replace = '';
```
Comment (ErikBoesen) [inline script.js]: Pretty sure you don't need a `replace` variable at all.

File: script.js (hunk 3 above)
Comment (AndrewLester) [reply script.js]: I don't think I can use "of" when I loop through the inputs variable because it's an object that isn't iterable.

Comment (ErikBoesen) [reply script.js]: Aren't they both arrays?

Comment (AndrewLester) [reply script.js]: For some reason it says inputs is an object. I think it's because of its instantiation as inputs = {};

File: script.js [hunk 5]
```diff
@@ -80,6 +80,13 @@ pg.submit.onclick = function() {
 
 function write(match) {
     var data = (fs.existsSync(path) && fs.statSync(path).size > 0) ? JSON.parse(fs.readFileSync(path)) : [];
+    for (var i = 0; i < data.length; i++) {
+        if (data[i].match === match.match && data[i].team === match.team) {
+            if (confirm('Do you want to replace the previous data for this match?')) {
```
Comment (ErikBoesen) [inline script.js]: You could put the confirm into the same if. Whatever though.

**Pull request merged**